llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -1517,10 +1518,28 @@ SDValue SelectionDAGLegalize::ExpandInsertToVectorThroughStack(SDValue Op) {
                      BaseVecAlignment);
 }
 
+SDValue SelectionDAGLegalize::ExpandConcatVectors(SDNode *Node) {
+  assert(Node->getOpcode() == ISD::CONCAT_VECTORS && "Unexpected opcode!");
+  SDLoc Dl(Node);
+  SmallVector<SDValue, 16> Ops;
+  unsigned NumOperands = Node->getNumOperands();
+  MVT VectorIdxType = TLI.getVectorIdxTy(DAG.getDataLayout());
+  for (unsigned I = 0; I < NumOperands; ++I) {
+    SDValue SubOp = Node->getOperand(I);
+    EVT VectorValueType =
+        SubOp->getValueType(0);
```

Comment:
```suggestion
    EVT VectorValueType = SubOp->getValueType(0);
```

Reply:
You already have the SDValue, you can access its value type directly:
```
EVT VectorValueType = SubOp.getValueType();
```